packages/web-components/@sfx/sayt/test/sayt.test.ts

```diff
@@ -0,0 +1,73 @@
+import { expect, spy, stub } from './utils';
+import Sayt from '../src/sayt';
+import { SAYT_EVENT } from '../src/events';
+
+describe('Sayt Component', () => {
+  let sayt;
+
+  beforeEach(() => {
+    sayt = new Sayt();
+  });
+
+  describe('connectedCallback()', () => {
+    it('should register event listeners to the window', () => {
+      const addEventListener = stub(window, 'addEventListener');
+
+      sayt.connectedCallback();
+
+      expect(addEventListener).to.be.calledWith(SAYT_EVENT.SAYT_SHOW, sayt.handleVisibilityEvent);
+      expect(addEventListener).to.be.calledWith(SAYT_EVENT.SAYT_HIDE, sayt.handleVisibilityEvent);
+    });
+  });
+
+
+  describe('disconnectedCallback()', () => {
+    it('should remove event listeners on the window', () => {
+      const removeEventListener = stub(window, 'removeEventListener');
+
+      sayt.disconnectedCallback();
+
+      expect(removeEventListener).to.be.calledWith(SAYT_EVENT.SAYT_SHOW, sayt.handleVisibilityEvent);
+      expect(removeEventListener).to.be.calledWith(SAYT_EVENT.SAYT_HIDE, sayt.handleVisibilityEvent);
+    });
+  });
+
+  describe('updated()', () => {
+    it('should change the hidden property if the show property has changed', () => {
+      sayt.hidden = true;
+      sayt.show = true;
+
+      sayt.updated(new Map([['show', false]]));
+
+      expect(sayt.hidden).to.equal(false);
+    });
+  });
+
+  describe('handleVisibilityEvent()', () => {
+    it('should set the show prop to true when receiving a SAYT_SHOW event', () => {
+      const dispatchedEvent = new CustomEvent(SAYT_EVENT.SAYT_SHOW, { detail: 'test' });
+
+      sayt.handleVisibilityEvent(dispatchedEvent);
+
+      expect(sayt.show).to.equal(true);
+    });
+
+    it('should set the show prop to false when receiving a SAYT_HIDE event', () => {
+      sayt.hidden = false;
+      sayt.show = true;
+      const dispatchedEvent = new CustomEvent(SAYT_EVENT.SAYT_HIDE, { detail: 'test' });
```

Comment:
I would put declarations first before assignments (i.e. move this above the the previous two lines).

Reply:
Updated.